server/aws-lsp-codewhisperer/src/language-server/codeWhispererSecurityScanServer.ts

```diff
@@ -1,21 +1,40 @@
 import { Server } from '@aws/language-server-runtimes'
 import { CredentialsProvider } from '@aws/language-server-runtimes/out/features'
-import { SecurityScanRequestParams } from '@aws/language-server-runtimes/out/features/securityScan/securityScan'
 import { pathToFileURL } from 'url'
 import { CancellationToken, ExecuteCommandParams } from 'vscode-languageserver'
 import { ArtifactMap } from '../client/token/codewhispererbearertokenclient'
 import { CodeWhispererServiceToken } from './codeWhispererService'
 import { DependencyGraphFactory } from './dependencyGraph/dependencyGraphFactory'
-import { SecurityScanHandler } from './securityScan/securityScanHandler'
-import { SecurityScanResponseParams } from './securityScan/types'
+import { getSupportedLanguageId, supportedSecurityScanLanguages } from './languageDetection'
+import SecurityScanDiagnosticsProvider from './securityScan/securityScanDiagnosticsProvider'
+import { SecurityScanCancelledError, SecurityScanHandler } from './securityScan/securityScanHandler'
+import { SecurityScanRequestParams, SecurityScanResponseParams } from './securityScan/types'
 import { parseJson } from './utils'
+import { SecurityScanEvent } from './telemetry/types'
 
 export const SecurityScanServerToken =
     (service: (credentialsProvider: CredentialsProvider) => CodeWhispererServiceToken): Server =>
-    ({ credentialsProvider, workspace, logging, lsp }) => {
+    ({ credentialsProvider, workspace, logging, lsp, telemetry }) => {
         const codewhispererclient = service(credentialsProvider)
+        const diagnosticsProvider = new SecurityScanDiagnosticsProvider(lsp, logging)
+        const scanHandler = new SecurityScanHandler(codewhispererclient, workspace, logging)
         let jobStatus: string
-        const runSecurityScan = async (params: SecurityScanRequestParams) => {
+        const securityScanStartTime = performance.now()
```

Comment:
Why are we creating this constant here? Shouldn't this be created each time a security scan gets started and not when the server is initialized?

Reply:
yep you are right, these need to be moved inside runSecurityScan, will do

Reply:
updated